### PR TITLE
Update product-schema

### PR DESCRIPTION
**Pull request by salander**
Suggestion of the product schema using masterVariant as required node.

### CODE REVIEW

File: examples/product/minimum.json [hunk 1]
```diff
@@ -4,9 +4,8 @@
       "name": {
         "en": "My Product"
       },
-      "variants": [
-        { "sku": "123" }
-      ]
+      "masterVariant":
+      { "sku": "123" }
```
Comment (emmenko) [inline examples/product/minimum.json]: Use correct formatting

File: schema/product.schema.json [hunk 2]
```diff
@@ -1,7 +1,5 @@
 {
-  "$schema": "http://json-schema.org/draft-04/schema#",
   "title": "SPHERE.IO product schema",
-  "url": "http://dev.sphere.io/http-api-projects-products.html",
```
Comment (emmenko) [inline schema/product.schema.json]: Why did you delete those?

File: schema/product.schema.json [hunk 3]
```diff
@@ -27,15 +25,54 @@
...
+                    },
+                    "value": { }
+                  },
+                  "additionalProperties": false,
+                  "required": ["name"]
+                }
+              }
+            },
+            "required": ["sku"]
+          },
```
Comment (emmenko) [inline schema/product.schema.json]: You should extract the variant as a reusable type 
https://github.com/sphereio/sphere-json-schemas/tree/master/schema/types

File: schema/product.schema.json [hunk 4]
```diff
@@ -80,10 +117,9 @@
             "$ref": "/reference"
           }
         },
-        "required": ["name", "variants"]
+        "required": ["name", "masterVariant"]
```
Comment (emmenko) [inline schema/product.schema.json]: Actually `name` and `productType` are required. @hajoeichler ?

File: schema/product.schema.json [hunk 5]
```diff
@@ -80,10 +117,9 @@
...
       }
     }
   },
-  "additionalProperties": false,
```
Comment (emmenko) [inline schema/product.schema.json]: Why did you remove this?

File: examples/product/variants.json [hunk 6]
```diff
@@ -17,6 +17,18 @@
...
         },
+      "variants": [
+        {
+          "id": 2,
+          "sku": "123",
+          "attributes": [
+            {
+              "name": "bar",
+              "value": "foo"
+            }
```
Comment (emmenko) [inline examples/product/variants.json]: This won't work, variants must be unique (attributes + sku + images + prices)

File: schema/product.schema.json (hunk 2 above)
Comment (salander) [inline schema/product.schema.json]: accidentially - will restore

File: schema/product.schema.json (hunk 5 above)
Comment (salander) [inline schema/product.schema.json]: accidentially - will restore

File: schema/product.schema.json (hunk 3 above)
Comment (salander) [inline schema/product.schema.json]: its not possible to reuse master variant because its id must be 1, for other variants should be minimum = 2. i can do it for the other variants...

File: schema/product.schema.json [hunk 7]
```diff
@@ -1,7 +1,7 @@
 {
   "$schema": "http://json-schema.org/draft-04/schema#",
   "title": "SPHERE.IO product schema",
-  "url": "http://dev.sphere.io/http-api-projects-products.html",
+  "url": "ttp://dev.sphere.io/http-api-projects-products.html",
```
Comment (hajoeichler) [inline schema/product.schema.json]: `h` missing

File: schema/product.schema.json (hunk 3 above)
Comment (hajoeichler) [inline schema/product.schema.json]: It's possible by combinations. See http://spacetelescope.github.io/understanding-json-schema/reference/combining.html#allof
So, you use the base type for variants and add then the extra constraint for the id where you use the variants type.

File: schema/product.schema.json (hunk 4 above)
Comment (hajoeichler) [inline schema/product.schema.json]: Ye, we need productType too.

File: examples/product/variants.json (hunk 6 above)
Comment (hajoeichler) [inline examples/product/variants.json]: Right the combination of attributes/images/prices must be unique. The sku isn't taken into account here.

File: schema/product.schema.json (hunk 4 above)
Comment (emmenko) [inline schema/product.schema.json]: But not `masterVariant`. It's not required to create a product...or?

Comment (hajoeichler) [inline schema/product.schema.json]: Yes, you need the masterVariant.

Comment (emmenko) [inline schema/product.schema.json]: `masterVariant` is optional for the API. Or you want to have it in the schema as required?

Comment (hajoeichler) [inline schema/product.schema.json]: Right, just verified it with playground. So let's remove it here.

File: schema/product.schema.json (hunk 3 above)
Comment (salander) [inline schema/product.schema.json]: @hajoeichler i've tried this, but seems that with the combination thing all fields of the referenced type and also the extra constraint is required now.
<code>
          "masterVariant": {
            "allOf": [
              {"$ref": "/variants"},
              {"properties": {
                "id": {
                    "type": "integer",
                    "minimum": 1,
                    "maximum": 1
                  }
                }
              }
            ]
          } <code>
maybe you can help?
